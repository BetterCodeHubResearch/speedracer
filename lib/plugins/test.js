const path = require('path')

const { createDir } = require('../.internal/util')

export default (options) => (reports, file) => {
  reports.forEach(report => {
    const dirname = path.join(path.dirname(file), options.snapshotsDirname)
    const filename = path.join(dirname, `${report.metadata.id}.speedracer`)

    return loadReport(filename).then(
      oldReport => {
        process.exit(0)
      },
      () => createDir(dirname).then(
        () => writeFile(filename, race.report.serialize())
      )
    )
  })
}
