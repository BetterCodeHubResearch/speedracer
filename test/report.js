// Native
import zlib from 'zlib'

// Packages
import test from 'ava'
import pify from 'pify'

// Ours
import Report from '../lib/report'
import Trace from '../lib/trace'
const { readFile } = require('../lib/.internal/util')

const loadEvents = filename =>
  readFile(filename, null)
    .then(pify(zlib.gunzip))
    .then(JSON.parse)

test('report profiling', async t => {
  const events = await loadEvents('./test/fixtures/alternate-body-background.trace.gz')
  const report = new Report({}, new Trace(events))
  t.snapshot(report.profiling)
})

test('report fps', async t => {
  const events = await loadEvents('./test/fixtures/alternate-body-background.trace.gz')
  const report = new Report({}, new Trace(events))
  t.snapshot(report.fps)
})

test('report first paint', async t => {
  const events = await loadEvents('./test/fixtures/alternate-body-background.trace.gz')
  const report = new Report({}, new Trace(events))
  t.snapshot(report.firstPaint)
})
