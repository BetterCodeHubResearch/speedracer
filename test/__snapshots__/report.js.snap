// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report first paint 1`] = `undefined`;

exports[`report fps 1`] = `undefined`;

exports[`report profiling 1`] = `
Object {
  "categories": Object {
    "painting": 8.899000406265259,
    "rendering": 10.437999486923218,
    "scripting": 13.000999689102173,
  },
  "events": Object {
    "Animation Frame Fired": 8.126000165939331,
    "Composite Layers": 6.616000294685364,
    "JS Frame": 4.82599949836731,
    "Paint": 2.283000111579895,
    "Recalculate Style": 4.758999943733215,
    "Run Microtasks": 0.04900002479553223,
    "Update Layer Tree": 5.678999543190002,
  },
  "functions": Object {
    "CompositeLayers": 6.616000294685364,
    "FireAnimationFrame": 8.126000165939331,
    "Paint": 2.283000111579895,
    "RunMicrotasks": 0.04900002479553223,
    "UpdateLayerTree": 5.678999543190002,
    "UpdateLayoutTree": 4.758999943733215,
    "f:@": 0.12800002098083496,
    "f:render@24": 2.303999662399292,
    "f:requestAnimationFrame@": 1.7019997835159302,
    "f:send@": 0.026999950408935547,
    "f:startRun@24": 0.20200002193450928,
    "f:ws.onmessage@24": 0.4630000591278076,
  },
}
`;
